www/source/partials/docs/_bp-kubernetes.html.md.erb

```diff
@@ -1,25 +1,176 @@
+[Kubernetes](http://kubernetes.io/) is an open source container cluster manager that is available as a stand-alone platform or embedded in several distributed platforms including [Google's Container Engine](https://cloud.google.com/container-engine/), [AWS Elastic Kubernetes Service](https://aws.amazon.com/eks/), [Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service/), and [Red Hat OpenShift](https://openshift.com/). Chef Habitat and Kubernetes are complementary: Kubernetes focuses on providing a platform for deployment, scaling, and operations of application containers across clusters of hosts while Chef Habitat manages the build pipeline and lifecycle of those application containers.
+
+## Chef Habitat on Kubernetes
+
+Chef Habitat can export Docker containers that run on Kubernetes. To benefit from Chef Habitat's gossip ring features like Service Discovery (binds) and Secrets, you must have a Chef Habitat bastion pod. A bastion pod provides an [Initial Peer](/docs/best-practices/#robust-supervisor-networks) to all other pods.
+
+The bastion pod is deployed in a robust fashion with a Kubernetes Stateful Set, Persistent Volume and livenesss checking to ensure availability and ring data persistence. A Kubernetes Service is attached to the Stateful Set in order to make it discoverable with DNS. It is recommended to deploy one Service and Stateful Set combination per Namespace.
+
+### Deploy the Chef Habitat bastion on Kubernetes
+
+Complete examples may be downloaded from [this folder](https://github.com/habitat-sh/habitat/tree/master/www/source/docs/examples/kubernetes_hab_bastion/)
+
+To explain how this works, let's break down the hab-bastion.yaml file:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: hab-bastion
+spec:
+  ports:
+  - name: gossip-listener
+    protocol: UDP
+    port: 9638
+    targetPort: 9638
+  - name: http-gateway
+    protocol: TCP
+    port: 9631
+    targetPort: 9631
+  selector:
+    app: hab-bastion
+  clusterIP: None
+```
+
+This service definition creates a virtual IP (VIP), opening access to the Chef Habitat service that will run on the pod.
+- The habitat gossip port (9638/UDP) listener
+- The habitat http-gateway (9631/TCP) listener
+- the service name becomes available in DNS (as `hab-bastion` or `hab-bastion.namespace-name`, etc) so any pod can find it
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: hab-bastion
+spec:
+    spec:
+      securityContext:
+        fsGroup: 42
+```
 
-## Chef Habitat Operator
+This section sets a group ownership for the persistent volume mount point, so that the hab supervisor can write to it.  The habitat user by default has uid `42` and gid `42`
```

Comment:
Would be good to be be consistent on how we are referring to the Habitat components on this line - ie, "hab supervisor" and "habitat user" - perhaps refer to them as "the Habitat supervisor" and "the Habitat user"?